### PR TITLE
`Assessment`: Fix number of assessments shown in course management statistics

### PR DESCRIPTION
**Pull request by Strohgelaender**
#### General
- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [ ] I added multiple integration tests (Spring) related to the features (with a high test coverage).

### Motivation and Context
@Santia-go noticed that the course cards on the course management side show a too high number of submissions. Students who just started the exercise but did not submitt anything still show up here, even though these participations don't have anything to assess. When moving to the assessment dashboard, a lower number is shown here (as expected).

### Description
I looked into the calls to `/for-assessment-dashboard` and `/stats-for-management-overview`. Both used different method of the submission repository, with `countByExerciseIdSubmittedBeforeDueDate()` actually filtering out participations without any submissions.

### Steps for Testing
Prerequisites:
- 1 Instructor
- 3 Students
- 1 Exercise [of each type], manual assessment enabled and an assessment due date in the future

1. Both students start the exercise, but only one actually submits a solution
2. As an instructor, navigate to the course management page after the due date.
3. Make sure that the progress bar displaying the assessment status has a maxium of 1 submission to grade.
4. After the due date, add another (late) submission as the third student.
5. Make sure that the progress bar stil displays only one submission to assess.

#### Code Review
- [x] Review 1
- [ ] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage
unchanged

### Screenshots
old:
grafik

new: 
grafik

Assessment Dashboard (unchanged):
grafik

### CODE REVIEW

**Review by bassner — APPROVED** [review]
Tested on TS1; the number of assessments is shown correctly!
Code LGTM.

**Review by Santia-go — APPROVED** [review]
Tested on TS1
Working as expected
Course management view:
image
Assessment dashboard:
image

Just the submissions (2) are considered for the assessment counter,, to the number of students who started the exercise (3).

File: src/test/java/de/tum/in/www1/artemis/util/DatabaseUtilService.java [hunk 1]
```diff
@@ -3533,11 +3533,20 @@ public void checkFeedbackCorrectlyStored(List<Feedback> sentFeedback, List<Feedb
...
-        var studentParticipation = ModelFactory.generateStudentParticipation(InitializationState.INITIALIZED, textExercise, user);
+        StudentParticipation studentParticipation;
+        if (participant instanceof User user) {
+            studentParticipation = ModelFactory.generateStudentParticipation(InitializationState.INITIALIZED, textExercise, user);
+        }
+        else if (participant instanceof Team team) {
+            studentParticipation = addTeamParticipationForExercise(textExercise, team.getId());
+        }
+        else {
+            throw new RuntimeException("unsupported participant!");
```
Comment (Dominik-Weinzierl) [inline src/test/java/de/tum/in/www1/artemis/util/DatabaseUtilService.java:3548]: Maybe we should stay consistent and use a capital letter at the beginning

**Review by Dominik-Weinzierl — APPROVED** [review]
Code looks good. Only one suggestion for the future 👍

**Review by mallgaier — APPROVED** [review]
Tested, works as expected. 
Good job 👍

File: src/test/java/de/tum/in/www1/artemis/util/DatabaseUtilService.java (hunk 1 above)
Comment (Strohgelaender) [reply src/test/java/de/tum/in/www1/artemis/util/DatabaseUtilService.java:3549]:
```suggestion
            throw new RuntimeException("Unsupported participant!");
```

**Pull request merged**